.github/workflows/doc-tests.yml

```diff
@@ -11,20 +11,48 @@ permissions:
 
 jobs:
   doc-test:
-    name: Sphinx-${{ matrix.builder }}
+    name: Sphinx Documentation Tests
     runs-on: ubuntu-latest
-    continue-on-error: ${{ matrix.experimental || false }}
-    strategy:
-      fail-fast: false
-      matrix:
-        builder: [ html, linkcheck ]
-        include:
-          # Run default html builder with warnings as error
-          - builder: html
-            args: -W
+
     steps:
       - uses: actions/checkout@v4
-      - name: Run sphinx builder ${{ matrix.builder }}
-        run: |
-          pip install hatch
-          hatch run docs:${{ matrix.builder }} ${{ matrix.args }}
+
+      - uses: actions/setup-python@v5
+        with:
+          python-version: '3.x'
+          cache: 'pip'
+      - name: Set PIPX_HOME
+        run: echo "PIPX_HOME=${{ github.workspace }}/.pipx" >> $GITHUB_ENV
+
+      - name: Cache pipx environments
+        uses: actions/cache@v3
+        with:
+          path: ${{ github.workspace }}/.pipx
+          # pipx is creating a cachedir.tag with unique
+          key: ${{ runner.os }}-pipx-home-${{ hashFiles('.pipx/.cache/CACHEDIR.TAG') }}
+          restore-keys: |
+            ${{ runner.os }}-pipx-home-
+
+
+
+      - name: Make sure latest hatch is installed
+        run: pipx upgrade hatch || pipx install hatch
+
+      - name: Cache Hatch environments
+        uses: actions/cache@v3
+        with:
+          path: ${{ github.workspace }}/.hatch_envs
+          # hatch envs are described in pyproject.toml
+          key: ${{ runner.os }}-hatch-envs-${{ hashFiles('pyproject.toml') }}
+          restore-keys: |
+            ${{ runner.os }}-hatch-envs-
+
+      - name: Set HATCH_ENV_TYPE_VIRTUAL_PATH
+        run: echo "HATCH_ENV_TYPE_VIRTUAL_PATH=${{ github.workspace }}/.hatch_envs" >> $GITHUB_ENV
```

Comment:
You should be able to place this in [`env` section](https://docs.github.com/en/actions/writing-workflows/workflow-syntax-for-github-actions#env)